### ISSUE TITLE
Cannot build with Visual Studio 2015

### ISSUE THREAD

**Issue by Niko-O** [dokan-dev/dokany#93]
I cloned the dokany repository, installed WDK 8.1 Update and opened the dokan.sln file in Visual studio 2015. The wiki page says to use Visual Studio 2013 but I don't have that available.
 When I first opened the project Visual Studio told me to upgrade some libraries and compilers because they "use an older version of the Visual C++-compilers and libraries". Because it warned me that I couldn't build the project without upgrading first I did that. The changes made can be viewed here:
https://github.com/Niko-O/dokany/commit/cc434a1f870dfb966a397e40d295011cb49bd81a
When I then go to Build -> Batch Build, select all the things and build the projects I get the following error:
* C2011 "timespec": "struct" Typneudefinition<br/>
in "dokany\dokan_fuse\include\fuse_win.h" line 61<br/>
Instance of this error: https://msdn.microsoft.com/en-us/library/sksadsda.aspx

And 15 of these warnings:
* C4305 "Typumwandlung": Verkürzung von "ULONG64" in "HANDLE"<br/>
in "dokany\dokan_mirror\mirror.c" line 362, 380, 418, ...<br/>
This happens every time something like ```(HANDLE)DokanFileInfo->Context``` occurs.<br/>
Instance of this warning: https://msdn.microsoft.com/en-us/library/0as1ke3f.aspx

And also 2 of these:
* C4201 "Nicht dem Standard entsprechende Erweiterung: Struktur/Union ohne Namen"<br/>
in "dokany\dokan_mirror\mirror.c" line 1143<br/>
and in "D:\Projects\Git\dokany\dokan_mount\mount.c" line 53<br/>
Instance of this warning: https://msdn.microsoft.com/en-us/library/1yex70d1.aspx

The warnings don't seem to matter that much but the error does. According to [this StackOverflow question](http://stackoverflow.com/questions/25713718/error-c2011-class-type-redefinition) the header file might be included multiple times which leads to multiple definitions of the struct. But the proposed solution is already in there:
```
#if ... !defined(_TIMESPEC_DEFINED)
#define _TIMESPEC_DEFINED
struct timespec { ... }
#endif
```
I have no idea what's going on.

**Comment by Liryna** [comment]
Hi @Niko-O,

Can you try the WDK10-VS2015 branch ? I am personnally using it with my vs2015.
It is made to use the wdk 10 and vs2015. (this branch is not always updated so merge the master in it to get the update)

**Comment by Niko-O** [comment]
@Liryna 
I have asked GitHub support how to set up my local repository in such a way that all files are as specified in a specific branch (using GitHub Desktop). I have not received a response yet so I'll just ask here.
How do I use the WDK10-VS2015 branch? If necessary I'll use the console to set it up.

**Comment by Liryna** [comment]
Sorry @Niko-O I never used GitHub Desktop.
But for the command lines:
```
cd dokany
git checkout -b WDK10-VS2015 //move to WDK10-VS2015
git pull master //pull master into your WDK10-VS2015 branch
```

**Comment by Niko-O** [comment]
```
D:\Projects\Git\dokany [WDK10-VS2015]> git pull master
fatal: 'master' does not appear to be a git repository
fatal: Could not read from remote repository.
Please make sure you have the correct access rights and the repository exists.
```

**Comment by Liryna** [comment]
and `git pull origin master`?

**Comment by Niko-O** [comment]
```
D:\Projects\Git\dokany [WDK10-VS2015]> git pull origin master
From https://github.com/Niko-O/dokany
 * branch            master     -> FETCH_HEAD
Already up-to-date.
```
I tried to build again (Visual Studio asked me to update the libraries and stuff again, which I did) but it resulted in the same errors.

**Comment by Liryna** [comment]
@Niko-O , I have updated the WDK10-VS2015 with the master for you.
Update your repository and use directly the WDK10-VS2015 branch.

**Comment by Niko-O** [comment]
Ok, at first I tried the following command after I deleted my fork and local repository:
```
git clone -b WDK10-VS2015 --single-branch https://github.com/dokan-dev/dokany.git
```
Building worked fine after that (and it didn't ask me to update the compiler).
Then I forked again (this way I can be sure that my fork is in the same state as the original repository), deleted my local repository again and did this:
```
git clone -b WDK10-VS2015 --single-branch https://github.com/Niko-O/dokany.git
```
Building also worked fine with that.
Thank you for your help.
... now on to Dokan-DotNet.
